src/app/prevention/strides-against/strides-against.component.ts

```diff
@@ -0,0 +1,260 @@
+
+  /**
+   * Called after the man has moved linearly to the desired position
+   */
+  handleQuestion(): void {
+    let currentImage = 0;
+    let index = this.pos - 1; // Finding which object eventually landed at the first position and so on...
+    for (let i = 0; i < this.countImage; i++) {
+      if (this.paperConfig[i].imagePositionIndex === this.sortedPosition[index].index) {
+        currentImage = i;
+        break;
+      }
+    }
+    index = currentImage;
+    this.elements[index].hide(); // Hiding the objects upon Man's arrival
+    this.question = this.ObjectsConfig[index].element.question;
+    const answer = this.ObjectsConfig[index].element.answer;
+    const image = this.ObjectsConfig[index].element.messageImageURL;
+    const message = this.ObjectsConfig[index].element.messageBody;
+    const dashboardLinkMessage = this.language.messages.failure.dashboardLinkMessage;
+    const messageSuccessBody = '<img src = "' + image + '" width=auto height=200px><br><br>' + message;
+    const messageFailBody = '<img src = "' + image + '" width=auto height=200px><br><br>' + message + '' +
+      '<br><br><a href="/"> ' + dashboardLinkMessage + ' </a>';
+
+    this.openDialog().subscribe(res => {
+      if (this.result === answer) {
+        swal(this.language.messages.success.title, messageSuccessBody, 'success');
+      } else {
+        this.man.animate({transform: 'r90'}, 2000, 'linear', () => {
+          swal({
+            type: 'error',
+            title: this.language.messages.failure.title,
+            html: messageFailBody,
+            confirmButtonText: this.language.messages.failure.confirmButtonText,
+          }).then(() => location.reload());
+          this.man.toBack();
+        })
+      }
+    });
+  }
+
+  /**
+   * Opens Dialog for asking questions when called
+   */
+  openDialog(): Observable<boolean> {
+    const dialogRef = this.dialog.open(DialogComponent, {
+      data: {
+        question: this.question,
+      },
+      position: {
+        top: '300px',
+        left: '560px'
+      },
+      disableClose: true,
+      autoFocus: true,
+      width: '482px'
+    });
+    dialogRef.afterClosed().subscribe(res => this.result = res);
+    return dialogRef.afterClosed();
+  }
+
+  /**
+   * Checks for the sign of the movement of amn in X direction
+   * @param  {number} position final position of the man.
+   */
+  checkXMoveSign(position: number): number {
+    if (position <= 8 || (position <= 24 && position > 17)) {
+      return 1
+    }
+    if ((position > 9 && position <= 16) || (position <= 32 && position > 25)) {
+      return -1
+    }
+  }
+
+  /**
+   * Checks if the the man's movement would involve Y direction
+   * @param  {number} initial Initial position of the man.
+   * @param  {number} final   Final position of the man.
+   */
+  checkYMove(initial: number, final: number): boolean {
+    return ((initial <= 8 && final > 8) || (initial <= 16 && final > 16) || (initial <= 24 && final > 24));
```

Comment:
this can also be made more efficient.. think a little about this.

Reply:
@shreyans29 As this was being used only in one place, So I used it in the "inline" way.